test/hpke_test.c

```diff
@@ -0,0 +1,1801 @@
+                        || hpke_mode == OSSL_HPKE_MODE_PSKAUTH) {
+                        /* check a borked p256 key */
+                        if (hpke_suite.kem_id == OSSL_HPKE_KEM_ID_P256) {
+                            /* set to fail decode of authpub this time */
+                            authpub[0] = 0x05;
+                            if (!TEST_true(OSSL_HPKE_CTX_set1_authpub(rctx,
+                                                                      authpub,
+                                                                      authpublen
+                                                                      )))
+                                overallresult = 0;
+                            if (!TEST_false(OSSL_HPKE_decap(rctx, senderpub,
+                                                            senderpublen,
+                                                            privp,
+                                                            infop, infolen)))
+                                overallresult = 0;
+                            /* fix authpub for real go at it */
+                            authpub[0] = 0x04;
+                        }
+                        if (!TEST_true(OSSL_HPKE_CTX_set1_authpub(rctx,
+                                                                  authpubp,
+                                                                  authpublen)))
+                            overallresult = 0;
+                    }
+                    if (startseq != 0) {
+                        if (!TEST_true(OSSL_HPKE_CTX_set_seq(rctx, startseq)))
+                            overallresult = 0;
+                    }
+                    if (!TEST_true(OSSL_HPKE_decap(rctx, senderpub,
+                                                   senderpublen, privp,
+                                                   infop, infolen)))
+                        overallresult = 0;
+                    /* throw in a call with a too-short clearlen */
+                    clearlen = 15;
+                    if (!TEST_false(OSSL_HPKE_open(rctx, clear, &clearlen,
+                                                   aadp, aadlen, cipher,
+                                                   cipherlen)))
+                        overallresult = 0;
+                    /* fix up real clearlen again */
+                    clearlen = OSSL_HPKE_TSTSIZE;
+                    if (!TEST_true(OSSL_HPKE_open(rctx, clear, &clearlen,
+                                                  aadp, aadlen, cipher,
+                                                  cipherlen)))
+                        overallresult = 0;
+                    OSSL_HPKE_CTX_free(rctx);
+                    EVP_PKEY_free(privp);
+                    privp = NULL;
+                    /* check output */
+                    if (!TEST_mem_eq(clear, clearlen, plain, plainlen)) {
+                        overallresult = 0;
+                    }
+                    if (privp) {
```

Comment:
dead code block, as you call `EVP_PKEY_free(privp);` 5 lines above

Reply:
true; deleted